internals/cli/cli.go

```diff
@@ -300,7 +302,15 @@ func Run(options *RunOptions) error {
 		return fmt.Errorf("cannot create client: %v", err)
 	}
 
-	parser := Parser(cli)
+	pebbleDir := options.PebbleDir
+	if pebbleDir == "" {
+		pebbleDir, _ = getEnvPaths()
+	}
+	parser := Parser(cli, &RunOptions{
+		ClientConfig: config,
+		Logger:       log,
+		PebbleDir:    pebbleDir,
```

Comment:
This applies the defaults generated in `Run()` (default clientConfig if not supplied, default logger if not supplied, default PebbleDir if not supplied) so that `Parser` (and in turn `CmdOptions`) can rely on all the fields being set to something.